compiler/rustc_hir_typeck/src/fallback.rs

```diff
@@ -464,6 +472,46 @@ impl<'tcx> FnCtxt<'_, 'tcx> {
         diverging_fallback
     }
 
+    fn lint_obligations_broken_by_never_type_fallback_change(
+        &self,
+        behavior: DivergingFallbackBehavior,
+        diverging_vids: &[TyVid],
+    ) {
+        let DivergingFallbackBehavior::FallbackToUnit = behavior else { return };
+
+        // Returns errors which happen if fallback is set to `fallback`
+        let try_out = |fallback| {
+            self.probe(|_| {
+                let obligations = self.fulfillment_cx.borrow().pending_obligations();
+                let ocx = ObligationCtxt::new(&self.infcx);
+                ocx.register_obligations(obligations.iter().cloned());
+
+                for &diverging_vid in diverging_vids {
+                    let diverging_ty = Ty::new_var(self.tcx, diverging_vid);
+
+                    _ = ocx.eq(&ObligationCause::dummy(), self.param_env, diverging_ty, fallback);
```

Comment:
```suggestion
                    let () = ocx.eq(&ObligationCause::dummy(), self.param_env, diverging_ty, fallback).expect("expected diverging var to be unconstrained");
```

Reply:
why `let () =`?

Reply:
because it doesn't implicitly assume the return type of `eq` is `Result<(), _>`

Reply:
either that, or just `ocx.eq().expect` without the `let ()`